fhi_users/emails.py

```diff
@@ -63,16 +66,34 @@ def send_verification_email(request, user: "User") -> None:
     current_site = get_current_site(request)
     mail_subject = "Activate your account."
     verification_token = default_token_generator.make_token(user)
-    VerificationToken.objects.create(user=user, token=verification_token)
-    params = urlencode({"token": verification_token, "uid": user.pk})
-    activation_link = f"https://www.fightpaperwork.com/activate-account/?{params}"
-    send_fallback_email(
-        mail_subject,
-        "acc_active_email",
-        {
-            "user": user,
-            "domain": current_site.domain,
-            "activation_link": activation_link,
-        },
-        user.email,
+    
+    try:
+        VerificationToken.objects.create(user=user, token=verification_token)
+    except Exception as e:
+        logger.error(f"Failed to create verification token for user {user.pk}: {e}")
+        raise Exception(f"Could not create verification token for user {user.pk}") from e
+
+    activation_link = (
+        "https://www.fightpaperwork.com/activate-account/?token={}&uid={}".format(
+            verification_token, user.pk
+        )
     )
+
+    try:
+        send_fallback_email(
+            mail_subject,
+            "acc_active_email",
+            {
+                "user": user,
+                "domain": current_site.domain,
+                "activation_link": activation_link,
+            },
+            user.email,
+        )
+        logger.info(f"Verification email sent to {user.email}")
+    except (SMTPException, ImproperlyConfigured) as e:
+        logger.error(f"Failed to send verification email to {user.email}: {e}")
+        raise
+    except Exception as e:
+        logger.error(f"An unexpected error occurred when sending email to {user.email}: {e}")
+        raise
```

Comment:
this is a weird place to log this, if we want to log e-mail errors put it in the e-mail utils.